File: src/main/java/org/cloudio/morpheus/mail/EmployeeUserMail.java
package org.cloudio.morpheus.mail;

/**
 * Created by zslajchrt on 24/08/15.
 */
public class EmployeeUserMail extends DefaultUserMail {
    public EmployeeUserMail(EmployeeAdapter employee) {
        super(employee);
    }

    @Override
    public void sendEmail(Message message) {
        Employee employee = ((EmployeeAdapter) getMailOwner());
        String signature = "\n\n" + employee.getFirstName() + " " + employee.getLastName() + "\n" + employee.getDepartment();

        Message newMsg = new Message(message);
        newMsg.setBody(message.getBody() + signature);

        super.sendEmail(newMsg);
    }

}

File: src/main/scala/org/cloudio/morpheus/mail/traditional/EmployeeUserMail.scala
package org.cloudio.morpheus.mail.traditional

/**
 * Created by zslajchrt on 24/08/15.
 */
trait EmployeeUserMail extends UserMail {
  this: Employee =>

  abstract override def sendEmail(message: Message): Unit = {
    val signature: String = "\n\n" + firstName + " " + lastName + "\n" + department

    super.sendEmail(message.copy(body = message.body + signature))
  }
}
